build/nv.d3.js

```diff
@@ -9593,179 +9603,1064 @@ nv.models.multiChart = function() {
+            var dataStack2 = data.filter(function(d) {return d.type == 'area' && d.yAxis == 2});
 
+            // Display noData message if there's nothing to show.
+            if (!data || !data.length || !data.filter(function(d) { return d.values.length }).length) {
+                nv.utils.noData(chart, container);
+                return chart;
+            } else {
+                container.selectAll('.nv-noData').remove();
+            }
 
-    //Create methods to allow outside functions to highlight a specific bar.
-    chart.highlightPoint = function(pointIndex, isHoverOver) {
-        chart.clearHighlights();
+            var series1 = data.filter(function(d) {return !d.disabled && d.yAxis == 1})
+                .map(function(d) {
+                    return d.values.map(function(d,i) {
+                        return { x: getX(d), y: getY(d) }
+                    })
+                });
+
+            var series2 = data.filter(function(d) {return !d.disabled && d.yAxis == 2})
+                .map(function(d) {
+                    return d.values.map(function(d,i) {
+                        return { x: getX(d), y: getY(d) }
+                    })
+                });
+    
+            x.domain(d3.extent(d3.merge(series1.concat(series2)), function(d) { return getX(d) }))
+                .range([0, availableWidth]);
+
+            x2.domain(x.domain()).range([0, availableWidth]);
+
+            var wrap = container.selectAll('g.nv-wrap.nv-multiChart').data([data]);
+            var gEnter = wrap.enter().append('g').attr('class', 'nv-wrap nvd3 nv-multiChart').append('g');
+            var g = wrap.select('g');
+
+            gEnter.append('g').attr('class', 'nv-focus');
+            gEnter.append('g').attr('class', 'nv-x nv-axis');
+            gEnter.append('g').attr('class', 'nv-y1 nv-axis');
+            gEnter.append('g').attr('class', 'nv-y2 nv-axis');
+            gEnter.append('g').attr('class', 'stack1Wrap');
+            gEnter.append('g').attr('class', 'stack2Wrap');
+            gEnter.append('g').attr('class', 'bars1Wrap');
+            gEnter.append('g').attr('class', 'bars2Wrap');
+            gEnter.append('g').attr('class', 'scatters1Wrap');
+            gEnter.append('g').attr('class', 'scatters2Wrap');
+            gEnter.append('g').attr('class', 'lines1Wrap');
+            gEnter.append('g').attr('class', 'lines2Wrap');
+            gEnter.append('g').attr('class', 'legendWrap');
+            gEnter.append('g').attr('class', 'nv-interactive');
+
+            var svg = d3.select(this.parentNode)
+                        .append('svg') //NOTE appending 'svg' is happening each time we use the legend
+                        .attr('width', availableWidth + margin2.left + margin2.right)
+                        .attr('height', availableHeight2 + margin2.top + margin2.bottom)
+                        // .attr('transform', 'translate(' + availableHeight + ',0)')
+                    ;
```

Comment:
Append a 'g' element to the existing svg instead of appending a whole new svg for the context.